extensions/libarchive/MergeContent.cpp

```diff
@@ -271,6 +281,16 @@ bool MergeContent::processBin(core::ProcessContext *context, core::ProcessSessio
       return false;
     }
     session->putAttribute(mergeFlow, BinFiles::FRAGMENT_COUNT_ATTRIBUTE, std::to_string(bin->getSize()));
+
+    std::unique_ptr<AttributeMerger> attributeMerger;
+    if (attributeStrategy_ == ATTRIBUTE_STRATEGY_KEEP_COMMON)
+      attributeMerger = std::unique_ptr < AttributeMerger > (new KeepOnlyCommonAttributesMerger(session, bin->getFlowFile(), mergeFlow));
+    else if (attributeStrategy_ == ATTRIBUTE_STRATEGY_KEEP_ALL_UNIQUE)
+      attributeMerger = std::unique_ptr < AttributeMerger > (new KeepAllUniqueAttributesMerger(session, bin->getFlowFile(), mergeFlow));
+    else
+      return false;
```

Comment:
Maybe we should [log something](https://xkcd.com/2200/) here.

Reply:
I think throwing would be a better way in this scenario as this should be unreachable because of the allowed values, so it would suggest a bigger problem. Same for the merge strategy, although there is an actual unreachable code there as `mergeFormat_` is checked both on line 258 and 260 so we could never reach the `return false` statement on line 267.